pkg/prometheus/promcfg.go

```diff
@@ -957,6 +957,7 @@ func (cg *ConfigGenerator) generateProbeConfig(
 		// Add configured relabelings.
 		xc := labeler.GetRelabelingConfigs(m.TypeMeta, m.ObjectMeta, m.Spec.Targets.StaticConfig.RelabelConfigs)
 		relabelings = append(relabelings, generateRelabelConfig(xc)...)
+		relabelings = generateAddressShardingRelabelingRulesForProbes(relabelings, shards)
```

Comment:
Do we want to configure sharing on every occasion? Or only when sharding is enabled in Prometheus?

Reply:
thanks for the suggestion. this is the change [e4ded8d](https://github.com/prometheus-operator/prometheus-operator/pull/5735/commits/e4ded8d7fd91d4d9874cb87a4cb88c1e31f1b2a2)